miniserver/abi/MiniWallet.json

```diff
@@ -1,1046 +1,1055 @@
-    "stateMutability": "nonpayable",
-    "type": "function"
-  },
-  {
-    "inputs": [
-      {
-        "internalType": "uint256",
-        "name": "amount",
-        "type": "uint256"
-      },
-      {
-        "internalType": "address",
-        "name": "from",
-        "type": "address"
-      },
-      {
-        "internalType": "address",
-        "name": "to",
-        "type": "address"
-      }
-    ],
-    "name": "send",
-    "outputs": [],
-    "stateMutability": "nonpayable",
-    "type": "function"
-  },
-  {
-    "inputs": [
-      {
-        "internalType": "bytes4",
-        "name": "interfaceId",
-        "type": "bytes4"
-      }
-    ],
-    "name": "supportsInterface",
-    "outputs": [
-      {
-        "internalType": "bool",
-        "name": "",
-        "type": "bool"
-      }
-    ],
-    "stateMutability": "view",
-    "type": "function"
-  },
-  {
-    "inputs": [
-      {
-        "internalType": "uint256",
-        "name": "amount",
-        "type": "uint256"
-      },
-      {
-        "internalType": "enum Enums.TokenType",
-        "name": "tokenType",
-        "type": "uint8"
-      },
-      {
-        "internalType": "uint256",
-        "name": "tokenId",
-        "type": "uint256"
-      },
-      {
-        "internalType": "address",
-        "name": "tokenAddress",
-        "type": "address"
-      },
-      {
-        "internalType": "address",
-        "name": "from",
-        "type": "address"
-      },
-      {
-        "internalType": "address",
-        "name": "to",
-        "type": "address"
-      }
-    ],
-    "name": "transfer",
-    "outputs": [],
-    "stateMutability": "nonpayable",
-    "type": "function"
-  },
-  {
-    "inputs": [
-      {
-        "internalType": "address",
-        "name": "newImplementation",
-        "type": "address"
-      }
-    ],
-    "name": "upgradeTo",
-    "outputs": [],
-    "stateMutability": "nonpayable",
-    "type": "function"
-  },
-  {
-    "inputs": [
-      {
-        "internalType": "address",
-        "name": "newImplementation",
-        "type": "address"
-      },
-      {
-        "internalType": "bytes",
-        "name": "data",
-        "type": "bytes"
-      }
-    ],
-    "name": "upgradeToAndCall",
-    "outputs": [],
-    "stateMutability": "payable",
-    "type": "function"
-  },
-  {
-    "inputs": [
-      {
-        "internalType": "address",
-        "name": "",
-        "type": "address"
-      }
-    ],
-    "name": "userBalances",
-    "outputs": [
-      {
-        "internalType": "uint256",
-        "name": "",
-        "type": "uint256"
-      }
-    ],
-    "stateMutability": "view",
-    "type": "function"
-  },
-  {
-    "inputs": [
-      {
-        "internalType": "uint256",
-        "name": "amount",
-        "type": "uint256"
-      }
-    ],
-    "name": "withdraw",
-    "outputs": [],
-    "stateMutability": "nonpayable",
-    "type": "function"
-  }
-],
\ No newline at end of file
+{
+  "_format": "hh-sol-artifact-1",
```

Comment:
wrong content. This should be abi only

Reply:
The code was written to use the full hardhat generated json and take the abi from it see [here](https://github.com/polymorpher/sms-wallet/blob/ws-miniwallet-v0.1/miniserver/blockchain.js#L28)
i.e. `miniWallet = new ethers.Contract(networkConfig.miniWalletAddress, MiniWallet.abi, provider)`
We can change to use abi only however when branching from main we received the error
```
UncaughtException [SyntaxError: /Users/john/one-wallet/sms-wallet/miniserver/abi/MiniWallet.json: Unexpected token , in JSON at position 20047].
node:internal/process/per_thread:221
      throw errnoException(err, 'kill');
      ^
```
If we prefer to use abi only need to ensure that it's JSON compliant and/or update the retrieval mechanism above.
Also should automate population of this folder when doing a `yarn compile` in `miniwallet`

Reply:
Have updated the logic to use abi only
abi's are populated when compiling using [hardhat-abi-exporter](https://www.npmjs.com/package/hardhat-abi-exporter)
The configuration is in `hardhat.config.ts`
```
  abiExporter: {
    path: '../miniserver/abi',
    runOnCompile: true,
    clear: true,
    flat: true,
    only: [':MiniWallet$', ':MiniProxy$'],
    format: 'json',
    spacing: 2
  },
```
Note: considered moving the abi's under a shared folder however left as miniserver only for now.

Reply:
there is an extra `,` which I forgot to remove, just remove from the end of the JSON